test/integration/targets/pip/tasks/url_packages.yml

```
@@ -0,0 +1,179 @@
+- name: Create local package
+  vars:
+    repo_path: "{{ remote_tmp_dir }}/pip_vcs_pkgs"
+  block:
+  - name: Create a nested package directory layout
+    file:
+      path: "{{ repo_path }}/dummy-root/dummy-subdirectory"
+      recurse: yes
+      state: directory
+      mode: '0755'
+
+  - name: Create a root-level pyproject.toml
+    copy:
+      dest: "{{ repo_path }}/pyproject.toml"
+      content: |
+        [build-system]
+        requires = ["setuptools>=61.0"]
+        build-backend = "setuptools.build_meta"
+
+        [project]
+        name = "dummy-root"
+        version = "0.0.1"
+        dependencies = []
+
+  - name: Create nested package's pyproject.toml
+    copy:
+      dest: "{{ repo_path }}/dummy-root/dummy-subdirectory/pyproject.toml"
+      content: |
+        [build-system]
+        requires = ["setuptools>=61.0"]
+        build-backend = "setuptools.build_meta"
+
+        [project]
+        name = "nested-package"
+        version = "0.0.1"
+        dependencies = []
+
+  - name: Create git repository
+    command:
+      cmd: "{{ item }}"
+    args:
+      chdir: "{{ repo_path }}"
+    environment:
+      GIT_CONFIG_NOSYSTEM: 1
+    loop:
+      - git init .
+      - git add .
+      - git -c user.name="Ansible Test" -c user.email="ansible@ansible.ansible" commit -m "First commit"
+
+- name: Setup venv
+  block:
+    - name: Set venv path
+      set_fact:
+        venv_path: "{{ remote_tmp_dir }}/testvenv"
+
+    - name: Ensure fresh venv
+      file:
+        path: "{{ venv_path }}"
+        state: absent
+
+    - name: Create fresh venv with no packages
+      pip:
+        name: []
+        virtualenv: "{{ venv_path }}"
+
+    - name: Get the pip version from the venv
+      command: "{{ venv_path }}/bin/python -c 'import pip; print(pip.__version__)'"
```

A better/agnostic way for getting the version is reading it from installed metadata:
```suggestion
      command: >-
        {{ venv_path }}/bin/python -Ic
        'import importlib.metadata as meta; print(meta.version("pip"))'
```
*(https://packaging.python.org/en/latest/discussions/versioning/#accessing-version-information-at-runtime)*

---

Accepted, thanks for the link.